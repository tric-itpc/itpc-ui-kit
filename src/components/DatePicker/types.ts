export enum HORIZONTAL_POSITION_CALENDAR {
  CALCULATED = "calculated",
  CENTER = "center",
  LEFT = "left",
  RIGHT = "right",
}

export enum VERTICAL_POSITION_CALENDAR {
  BOTTOM = "bottom",
  TOP = "top",
}

export type PositionType = "absolute" | "fixed"

export enum ALLOWED_POSITIONS {
  ABSOLUTE = "absolute",
  FIXED = "fixed",
}

export interface InputDimensions {
  inputBottom: number
  inputHeight: number
  inputLeft: number
  inputTop: number
  inputWidth: number
}

export interface CalendarDimensions {
  calendarHeight: number
  calendarWidth: number
}

export interface DocumentDimensions {
  documentHeight: number
  documentWidth: number
}

export interface GetHorizontalPositionArg {
  calendarWidth: number
  distanceRight: number
  documentWidth: number
  inputLeft: number
  inputWidth: number
  scrollbarWidth: number
}

export interface AbsolutePositionArg {
  calendarHeight: number
  calendarWidth: number
  documentWidth: number
  inputHeight: number
  inputLeft: number
  inputWidth: number
}

export interface AbsolutePosition {
  absoluteBottomPosition: number
  absoluteCalculatedPosition: number
  absoluteCenterPosition: number
  absoluteDefault: number
  absoluteTopPosition: number
}
