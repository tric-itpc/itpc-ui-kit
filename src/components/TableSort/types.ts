export enum SortType {
  ASCENDING = "ascending",
  DESCENDING = "descending",
  NONE = "none",
}

export type RowType<T = object> = {
  [key in keyof T]: T[key]
} & {
  id: string
}

export type SorterFn<T> = (a: T, b: T) => number

export interface Column<T> {
  isSortable: boolean
  name: keyof T
  sorter?: SorterFn<T>
  title: string
}

export interface KeySort<T> {
  isSortable: boolean
  name: string
  order?: SortType
  sorter?: SorterFn<T>
}
