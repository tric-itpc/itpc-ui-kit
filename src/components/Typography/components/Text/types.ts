import { AllowedClassListProps } from "../types"

import { TextProps } from "./index"

export type ClassListProps = Pick<TextProps, AllowedClassListProps>
