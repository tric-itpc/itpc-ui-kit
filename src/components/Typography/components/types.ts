import { TextSize, TextType, TextWeight } from "../../../enums"

export type AllowedClassListProps =
  | "className"
  | "size"
  | "through"
  | "type"
  | "underline"
  | "weight"

export interface BaseClassListProps {
  className?: string
  size?: TextSize
  through?: boolean
  type?: TextType
  underline?: boolean
  weight?: TextWeight
}
