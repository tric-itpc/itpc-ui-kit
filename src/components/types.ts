import React from "react"

export type InputState = 'default' | 'cancel' | 'loading' | 'success' | 'warning'
export type InputType = 'password' | 'text'
export type ButtonType = 'button' | 'submit' | 'reset'
export type ButtonVariant = 'white' | 'red'
export type InputCheckboxType = "checkbox" | "radio"
export type Orientation = 'top' | 'right' | 'bottom' | 'left'
export type PopupSize = 'small' | 'normal'
export type PopupVariant = 'default' | 'error' | 'warning' | 'success'
export type PopupPosition =
  'top-left'
  | 'top-center'
  | 'top-right'
  | 'center-left'
  | 'center-center'
  | 'center-right'
  | 'bottom-left'
  | 'bottom-center'
  | 'bottom-right'
export type ValidationState = 'valid' | 'invalid'

export interface TabsItem {
  title: string,
  content: React.ReactElement<React.ReactNode>
}

export interface PaginationResult {
  currentPage?: number
  start: number
  end: number
}

export interface Item {
  id: string
  value: string
  disabled?: boolean
}

export interface FormattedValues {
  value: string,
  formattedValue: string
}
