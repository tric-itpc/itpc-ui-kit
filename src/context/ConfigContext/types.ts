import { Theme } from "../../enums"

export interface ThemeContext {
  setTheme: (theme: Theme) => void
  theme: Theme
  themeClass: string
}

export interface ConfigContextProps {
  themeComponent?: ThemeContext
}
