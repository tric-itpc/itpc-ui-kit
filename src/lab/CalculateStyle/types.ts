export type PositionType = "absolute" | "fixed"

export enum ALLOWED_POSITIONS {
  ABSOLUTE = "absolute",
  FIXED = "fixed",
}

export interface DocumentDimensions {
  documentHeight: number
  documentWidth: number
}

export interface ParentDimensions {
  parentBottom: number
  parentHeight: number
  parentLeft: number
  parentTop: number
  parentWidth: number
}

export interface ElementDimensions {
  elementHeight: number
  elementWidth: number
}

export enum HORIZONTAL_POSITION {
  CALCULATED = "calculated",
  CENTER = "center",
  LEFT = "left",
  RIGHT = "right",
}

export enum VERTICAL_POSITION {
  BOTTOM = "bottom",
  TOP = "top",
}

export interface GetHorizontalPositionArg {
  defaultParentWidth?: number
  distanceRight: number
  documentWidth: number
  elementWidth: number
  parentLeft: number
  parentWidth: number
  scrollbarWidth: number
}
