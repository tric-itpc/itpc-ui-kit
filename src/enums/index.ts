export enum UIKitColors {
  black = "#000",
  green = "#4DB04D",
  grey = "#B2B7C7",
  greyLight = "#E5E5E5",
  purple = "#5C53AC",
  purpleLight = "#6F65CB",
  red = "#D42564",
  redDark = "#C9184A",
  white = "#FFF",
  yellow = "#DCB21E",
}

export enum Theme {
  DARK = "dark",
  DEFAULT = "default",
}

export enum SortType {
  ASCENDING = "ascending",
  DESCENDING = "descending",
  NONE = "none",
}

export enum NumberColumns {
  ONE = "one",
  TWO = "two",
  ZERO = "zero",
}

export enum RowAlign {
  BOTTOM = "bottom",
  MIDDLE = "middle",
  TOP = "top",
}

export enum RowJustify {
  CENTER = "center",
  END = "end",
  SPACE_AROUND = "around",
  SPACE_BETWEEN = "between",
  SPACE_EVENLY = "evenly",
  START = "start",
}

export enum TextSize {
  L = "l",
  M = "m",
  S = "s",
  XL = "xl",
  XS = "xs",
  XXL = "xxl",
  XXS = "xxs",
}

export enum TextWeight {
  BOLD = "bold",
  LIGHT = "light",
  MEDIUM = "medium",
  NORMAL = "normal",
}

export enum TextType {
  DANGER = "danger",
  DISABLED = "disabled",
  PRIMARY = "primary",
  SECONDARY = "secondary",
  SUCCESS = "success",
  WARNING = "warning",
}

export enum TextTag {
  CODE = "code",
  EM = "em",
  PARAGRAPH = "p",
  STRONG = "strong",
}

export enum AnimationTypes {
  FIXED = "FIXED",
  TRANSFORM_BY_AXIS_X = "transform-by-axis-x",
  TRANSFORM_BY_AXIS_X_Y = "transform-by-axis-x_y",
}
