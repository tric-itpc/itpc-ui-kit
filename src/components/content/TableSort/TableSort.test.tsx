import React from "react"

import { fireEvent, render, screen, waitFor } from "@testing-library/react"

import { NumberColumns } from "../../../enums"

import { Column } from "./types"

import { TableSort } from "./index"

interface RowType {
  age: number
  id: string
  location: string
  name: string
}

const columns: Column<RowType>[] = [
  { isSortable: true, name: "name", title: "Name" },
  { isSortable: true, name: "age", title: "Age" },
  { isSortable: false, name: "location", title: "Location" },
]

const rows: RowType[] = [
  { age: 30, id: "1", location: "New York", name: "Alice" },
  { age: 25, id: "2", location: "Los Angeles", name: "Bob" },
  { age: 35, id: "3", location: "Chicago", name: "Charlie" },
]

describe("Компонент TableSort", () => {
  test("должен корректно рендериться", () => {
    const { container } = render(
      <TableSort columns={columns} rows={rows} title="Test Table" />
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  test("должен отображать заголовок таблицы", () => {
    render(<TableSort columns={columns} rows={rows} title="Test Table" />)
    const tableElement = screen.getByTitle("Test Table")
    expect(tableElement).toBeInTheDocument()
  })

  test("должен корректно отображать колонки", () => {
    render(<TableSort columns={columns} rows={rows} title="Test Table" />)
    expect(screen.getByText("Name")).toBeInTheDocument()
    expect(screen.getByText("Age")).toBeInTheDocument()
    expect(screen.getByText("Location")).toBeInTheDocument()
  })

  test("должен сортировать данные по возрастанию при клике на заголовок", () => {
    const columns = [
      { isSortable: true, name: "name", title: "Name" },
      {
        isSortable: true,
        name: "age",
        sorter: (a: RowType, b: RowType) => a.age - b.age,
        title: "Age",
      },
      { isSortable: true, name: "location", title: "Location" },
    ]

    const rows = [
      { age: 30, id: "1", location: "New York", name: "Alice" },
      { age: 25, id: "2", location: "Los Angeles", name: "Bob" },
      { age: 35, id: "3", location: "Chicago", name: "Charlie" },
    ]

    render(<TableSort columns={columns} rows={rows} title="Test Table" />)

    const ageHeader = screen.getAllByRole("columnheader")[1]
    const beforeSortedRows = screen.getAllByRole("row").slice(1)

    expect(beforeSortedRows[0].children[0].textContent).toBe("Alice")
    expect(beforeSortedRows[1].children[0].textContent).toBe("Bob")
    expect(beforeSortedRows[2].children[0].textContent).toBe("Charlie")

    expect(beforeSortedRows[0].children[1].textContent).toBe("30")
    expect(beforeSortedRows[1].children[1].textContent).toBe("25")
    expect(beforeSortedRows[2].children[1].textContent).toBe("35")

    fireEvent.click(ageHeader)

    const sortedRows = screen.getAllByRole("row").slice(1)

    waitFor(() => {
      expect(sortedRows[0].children[0].textContent).toBe("Bob")
      expect(sortedRows[1].children[0].textContent).toBe("Alice")
      expect(sortedRows[2].children[0].textContent).toBe("Charlie")

      expect(sortedRows[0].children[1].textContent).toBe("25")
      expect(sortedRows[1].children[1].textContent).toBe("30")
      expect(sortedRows[2].children[1].textContent).toBe("35")
    })
  })

  test("должен сортировать данные по убыванию при повторном клике на заголовок", () => {
    render(<TableSort columns={columns} rows={rows} title="Test Table" />)

    const ageColumn = screen.getByText("Age")
    fireEvent.click(ageColumn) // Сортировка по возрасту по возрастанию
    fireEvent.click(ageColumn) // Сортировка по возрасту по убыванию

    const rowsAfterSortDesc = screen.getAllByRole("row")

    waitFor(() => {
      expect(rowsAfterSortDesc[1].textContent).toContain("Charlie")
      expect(rowsAfterSortDesc[2].textContent).toContain("Alice")
      expect(rowsAfterSortDesc[3].textContent).toContain("Bob")
    })
  })

  test("должен корректно сортировать данные по двум колонкам", () => {
    const updatedColumns = [
      { isSortable: true, name: "name", title: "Name" },
      { isSortable: true, name: "age", title: "Age" },
    ]

    const updatedRows = [
      { age: "30", id: "1", location: "New York", name: "Alice" },
      { age: "25", id: "2", location: "Los Angeles", name: "Bob" },
      { age: "30", id: "3", location: "Chicago", name: "Charlie" },
    ]

    render(
      <TableSort
        columns={updatedColumns}
        rows={updatedRows}
        sortByNumberColumns={NumberColumns.TWO}
        title="Test Table"
      />
    )

    const nameColumn = screen.getByText("Name")
    fireEvent.click(nameColumn) // Сортировка по имени

    const rowsAfterSort = screen.getAllByRole("row")

    waitFor(() => {
      expect(rowsAfterSort[1].textContent).toContain("Alice")
      expect(rowsAfterSort[2].textContent).toContain("Bob")
      expect(rowsAfterSort[3].textContent).toContain("Charlie")
    })
  })
})
