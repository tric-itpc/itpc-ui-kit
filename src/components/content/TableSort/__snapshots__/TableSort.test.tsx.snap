// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Компонент TableSort должен корректно рендериться 1`] = `
<table
  class="itpc-table-sort"
  title="Test Table"
>
  <thead
    class="itpc-table-sort__head"
  >
    <tr>
      <th
        class="itpc-table-sort__head-no-sort"
      >
        <div
          class="itpc-table-sort__wrap-cell"
        >
          Name
        </div>
      </th>
      <th
        class="itpc-table-sort__head-no-sort"
      >
        <div
          class="itpc-table-sort__wrap-cell"
        >
          Age
        </div>
      </th>
      <th
        class="itpc-table-sort__head-no-sort"
      >
        <div
          class="itpc-table-sort__wrap-cell"
        >
          Location
        </div>
      </th>
    </tr>
  </thead>
  <tbody
    class="itpc-table-sort__body"
  >
    <tr
      class="itpc-table-sort__row"
    >
      <td
        class="itpc-table-sort__cell"
      >
        Alice
      </td>
      <td
        class="itpc-table-sort__cell"
      >
        30
      </td>
      <td
        class="itpc-table-sort__cell"
      >
        New York
      </td>
    </tr>
    <tr
      class="itpc-table-sort__row"
    >
      <td
        class="itpc-table-sort__cell"
      >
        Bob
      </td>
      <td
        class="itpc-table-sort__cell"
      >
        25
      </td>
      <td
        class="itpc-table-sort__cell"
      >
        Los Angeles
      </td>
    </tr>
    <tr
      class="itpc-table-sort__row"
    >
      <td
        class="itpc-table-sort__cell"
      >
        Charlie
      </td>
      <td
        class="itpc-table-sort__cell"
      >
        35
      </td>
      <td
        class="itpc-table-sort__cell"
      >
        Chicago
      </td>
    </tr>
  </tbody>
</table>
`;
